ibtk/include/ibtk/HierarchyIntegrator.h

```diff
@@ -697,6 +698,16 @@ class HierarchyIntegrator : public SAMRAI::mesh::StandardTagAndInitStrategy<NDIM
     void putToDatabase(SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> db) override;
 
 protected:
+    /*!
+     * Pure virtual function that integrates the hierarchy from current_time to new_time.
+     *
+     * Implementations of this virtual function are not required to synchronize
+     * data on the patch hierarchy.  Data synchronization may be done
+     * (optionally) in a specialization of the public virtual member function
+     * postprocessIntegrateHierarchy().
```

Comment:
Could you add a `@note` here explaining why we did this, i.e., some discussion of when callbacks are called?

Reply:
Could you also mention that derived classes should typically call the parent class `integrateHierarchySpecialized()` at the top of their own implementation of this function?